var/spack/repos/builtin/packages/hpcc/package.py

```diff
@@ -0,0 +1,208 @@
+
+    version('develop', hg=hg)
+    version('1.5.0', sha256='0a6fef7ab9f3347e549fed65ebb98234'
+                            'feea9ee18aea0c8f59baefbe3cf7ffb8', preferred=True)
+
+    variant(
+        'fft', default='internal', description='FFT library to use',
+        values=('internal', 'fftw2', 'mkl'), multi=False
+    )
+
+    depends_on('gmake', type='build')
+    depends_on('mpi@1.1:')
+    depends_on('blas')
+    depends_on('fftw-api@2+mpi', when='fft=fftw2')
+    depends_on('mkl', when='fft=mkl')
+
+    arch = '{0}-{1}'.format(platform.system(), platform.processor())
+
+    config = {
+        '@SHELL@': '/bin/sh',
+        '@CD@': 'cd',
+        '@CP@': 'cp',
+        '@LN_S@': 'ln -s',
+        '@MKDIR@': 'mkdir',
+        '@RM@': '/bin/rm -f',
+        '@TOUCH@': 'touch',
+        '@ARCHIVER@': 'ar',
+        '@ARFLAGS@': 'r',
+        '@RANLIB@': 'echo',
+        '@ARCH@': arch,
+        '@MPDIR@': '',
+        '@MPINC@': '',
+        '@MPLIB@': '',
+        '@F2CDEFS@': '-DAdd_ -DF77_INTEGER=int -DStringSunStyle',
+        '@LADIR@': '',
+        '@LAINC@': '',
+        '@LALIB@': '',
+        '@CC@': None,
+        '@CCNOOPT@': '',
+        '@CCFLAGS@': '-O3',
+        '@LINKER@': '$(CC)',
+        '@LINKFLAGS@': ''
+    }
+
+    def patch(self):
+        if 'fftw' in self.spec:
+            # spack's fftw2 prefix headers with floating point type
+            filter_file(r"^\s*#include <fftw.h>", "#include <sfftw.h>",
+                        "FFT/wrapfftw.h")
+            filter_file(r"^\s*#include <fftw_mpi.h>", "#include <sfftw_mpi.h>",
+                        "FFT/wrapmpifftw.h")
+
+    def _write_make_arch(self, spec, prefix):
+        """write make.arch file"""
+        with working_dir('hpl'):
+            # copy template make.arch file
+            make_arch_filename = 'Make.{0}'.format(self.arch)
+            copy(join_path('setup', 'Make.UNKNOWN.in'), make_arch_filename)
+
+            # fill template with values
+            make_arch = FileFilter(make_arch_filename)
+            for k, v in self.config.items():
+                make_arch.filter(k, v)
+
+    def edit(self, spec, prefix):
+        # Message Passing library (MPI)
+        self.config['@MPINC@'] = '-I{0}'.format(spec['mpi'].prefix.include)
+        self.config['@MPLIB@'] = '-L{0}'.format(spec['mpi'].prefix.lib)
+
+        lin_alg_libs = []
+        # FFT
+        if self.spec.variants['fft'].value in ('fftw2', 'mkl'):
+            self.config['@LAINC@'] += ' -DUSING_FFTW'
+
+            if self.spec.variants['fft'].value == 'fftw2':
+                self.config['@LAINC@'] += \
+                    ' -I{0}'.format(spec['fftw'].prefix.include)
+                # fftw does not set up libs for version 2
+                lin_alg_libs.append(
+                    join_path(spec['fftw'].prefix.lib, 'libsfftw_mpi.so'))
+                lin_alg_libs.append(
+                    join_path(spec['fftw'].prefix.lib, 'libsfftw.so'))
```

Comment:
You should use something like `spec['fftw-api'].libs` here.

Reply:
fftw-api@2 does not populate spec['fftw-api'].libs plus there are a lot of different versions based on type of floating point.